ivy/data_classes/array/array.py

```diff
@@ -431,8 +431,11 @@ def __getattribute__(self, item):
     def __getattr__(self, item):
         try:
             attr = self._data.__getattribute__(item)
-        except AttributeError:
-            attr = self._data.__getattr__(item)
+        except AttributeError as e:
```

Comment:
actually this is something that was being worked on in [this](https://github.com/unifyai/ivy/pull/16339) PR but ofc we could close that one now. @Infrared1029 could you please review this PR as the author of that task? Thanks @AnnaTz 😄